app/presenters/concerns/review_tasks/permitted_development_right_presenter.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module ReviewTasks
+  extend ActiveSupport::Concern
+
+  class PermittedDevelopmentRightPresenter
+    include Presentable
+
+    def initialize(template, planning_application)
+      @planning_application = planning_application
+      @template = template
+      @permitted_development_right = planning_application.permitted_development_right
+      @status = @permitted_development_right.review_status
+    end
+
+    def task_list_row
+      html = tag.span class: "app-task-list__task-name" do
+        concat permitted_development_right_link
+      end
+
+      html.concat permitted_development_rights_status_tag
+    end
+
+    private
+
+    attr_reader :status, :permitted_development_right
+
+    def permitted_development_right_link
+      link_to("Permitted development rights", permitted_development_right_link_url, class: "govuk-link")
+    end
+
+    def permitted_development_right_link_url
+      case status.humanize
+
+      when "Review not started"
+        edit_planning_application_review_permitted_development_right_path(planning_application,
+                                                                          permitted_development_right)
+      when "Review in progress", "Review complete"
+        planning_application_review_permitted_development_right_path(planning_application, permitted_development_right)
+      else
+        raise ArgumentError, "The status provided: '#{status}' is not valid"
+      end
+    end
+
+    def permitted_development_rights_status_tag
```

Comment:
I think we're stretching the role of the presenter too far when there's a better alternative. It's complicated to try to understand the HTML structure. The status tag component is offering an easier solution you can use the template and then the focus of the test can be the thing that changes ... in this case` govuk-tag--blue` and not the entire html structure that you end up having to test (It's right to test the constructed HTML but comparing the whole html line means you're not sure what is the thing that varies and is most likely to break in this test). 

This same, relatively, complicated code is present in [all the presenters with status tags](https://github.com/unboxed/bops/blob/main/app/presenters/concerns/assessment_tasks_presenter.rb). We haven't tried to abstract the status tag code and we should want to when it's used so often.

Have you tried the status tag component? What did you think of the abstraction?

There haven't been tests for the status tag as the rspec setup had been incorrectly configured. However, that's now fixed and this is an example test which will be in my PR:

```ruby
# frozen_string_literal: true

require "rails_helper"

RSpec.describe StatusTags::ReviewPolicyClassComponent, type: :component do
  it "displays start status" do
    review_policy_class = create(:review_policy_class, status: :not_checked_yet)

    render_inline(described_class.new(review_policy_class: review_policy_class))

    expect(page).to have_css ".govuk-tag--grey", text: "Not checked yet"
  end

  it "displays complete status" do
    review_policy_class = create(:review_policy_class, status: :complete)

    render_inline(described_class.new(review_policy_class: review_policy_class))

    expect(page).to have_css ".govuk-tag--blue", text: "Complete"
  end
end
```
This is **unreviewed** code that goes with the test class for refrence


```ruby
module StatusTags
  class ReviewPolicyClassComponent < StatusTags::BaseComponent
    def initialize(review_policy_class:)
      @review_policy_class = review_policy_class
    end

    private

    def status
      @review_policy_class&.status&.to_sym || :not_checked_yet
    end
  end
end
```

Reply:
Thanks for the suggestion, I was being consistent with previous work achieving this sort of thing but I forgot about this module. 
I think we could do some refactor work moving all the status html to this format if the team agree but perhaps as a separate PR.